nixos/modules/services/mail/roundcube.nix

```diff
@@ -0,0 +1,89 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.roundcube;
+in
+{
+  options.services.roundcube = {
+    enable = mkEnableOption "Roundcube";
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "[::]";
+      description = "Listening address. IPv6 addresses must be enclosed in square brackets";
+    };
+
+    listenPort = mkOption {
+      type = types.int;
+      default = 80;
+      description = "Listening port";
+    };
+    
+    subDomain = mkOption {
+      type = types.str;
+      example = "webmail";
+      description = "Sub-domain to use which is the name of the nginx vhost";
+    };
+    
+    extraConfig = mkOption {
+      type = types.str;
+      default = ''
+        <?php
```

Comment:
how about using `types.lines` rather than `types.str`? When you need to override a configuration parameter you'd currently have to copy the entire configuration into your nix expression.

Then you could do something like this in the `config` block:

``` nix
{
  environment.etc."roundcube/config.inc.php".text = ''
    <?php
    ${cfg.extraConfig}
  '';
}
```